src/pool/srv_target.c

```diff
@@ -496,14 +510,65 @@ ds_pool_put(struct ds_pool *pool)
 	ABT_mutex_unlock(pool_cache_lock);
 }
 
+#define MAX_HDL_BUF_SIZE	1024
+void
+pool_fetch_hdls_ult(void *data)
+{
+	struct ds_pool	*pool = data;
+	char		buf[MAX_HDL_BUF_SIZE];
+	d_iov_t		iov = { 0 };
+	int		rc = 0;
+
+	ABT_mutex_lock(pool->sp_mutex);
+	if (pool->sp_map == NULL)
```

Comment:
Could you add some comments here to explain why it needs to wait on 'sp_fetch_hdls_cond' when 'sp_map == NULL'? If fetching handles can only happen when sp_map changed from NULL to non-NULL, why don't we just fetch handles there? (so that this 'sp_fetch_hdls_cond' could be removed)

Reply:
well, each server only knows the pool leader, once sp_map is set, i.e. iv ns is setup. Sure, I will add some comments here.